File: server-aspnet-fastendpoints/src/Org.OpenAPITools/Models/SubscriptionPlan.cs
namespace Org.OpenAPITools.Models;


/// <summary>
/// 
/// </summary>
public class SubscriptionPlan 
{
    public long Id { get; set; }
    public bool Active { get; set; }
    public bool Valid { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public List<SubscriptionOption> Options { get; set; }
    public double Price { get; set; }
    public double Promo { get; set; }
    public double TransactionFee { get; set; }
    public string Role { get; set; }
    public bool DefaultPlan { get; set; }
    public bool Visible { get; set; }
    public Asset Image { get; set; }
}

File: server-scala-play-server/app/model/SubscriptionPlan.scala
package model

import play.api.libs.json._

/**
  * Represents the Swagger definition for SubscriptionPlan.
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
case class SubscriptionPlan(
  id: Option[Long],
  active: Option[Boolean],
  valid: Option[Boolean],
  name: Option[String],
  description: Option[String],
  options: Option[List[SubscriptionOption]],
  price: Option[Double],
  promo: Option[Double],
  transactionFee: Option[Double],
  role: Option[String],
  defaultPlan: Option[Boolean],
  visible: Option[Boolean],
  image: Option[Asset]
)

object SubscriptionPlan {
  implicit lazy val subscriptionPlanJsonFormat: Format[SubscriptionPlan] = Json.format[SubscriptionPlan]
}

